.travis.yml

```diff
@@ -78,6 +78,16 @@ jobs:
       script:
         - NODE=$(which node) make lint lint-py
 
+    - name: "Docs"
+      language: node_js
+      node_js: "node"
+      install:
+        - pyenv global 2.7.15
+        - make doc-only
+        - npm install -g linkinator
+      script:
+        - linkinator out/doc/api -r
```

Comment:
Maybe just scanning all.html is a good idea? Or does that just not make a difference?

Reply:
Can we make it only run when the docs are touched? (Like a make target that depends on docs?)

Reply:
Can we use the same mechanism that add doc label to PR ?

Reply:
Added a change that `grep`s the diff for `*.md` files in `doc/api` and only runs the scan if grep returns something.

For local files `linkinator` works on a directory level so to scan, say, just `all.html` would involve moving files into a temporary directory. If someone has time maybe they can run `linkinator` on just `all.html` and see how many links it scans.

Reply:
> 
> 
> Added a change that `grep`s the PR's patch diff for `*.md` files in `doc/api` and only runs the scan if grep returns something.
> 
> For local files `linkinator` works on a directory level so to scan, say, just `all.html` would involve moving files into a temporary directory. If someone has time maybe they can run `linkinator` on just `all.html` and see how many links it scans.